### ISSUE TITLE
Support for variable fonts?

### ISSUE THREAD

**Issue by doutatsu** [feat-agency/vite-plugin-webfont-dl#34]
I am using Inter CSS Variable font, so that I get access to everything in one go, as per documentation here - https://rsms.me/inter/#variable. I am unable to get it to work though, and the only way I can make it work is by going the Google Font route, meaning I have to pick which specific styles for this font to use, which is not ideal. 

Am I doing something wrong or Variable fonts are not supported? If the latter, is it possible to make it work?

**Comment by 0xb4lint** [comment]
Hi @doutatsu!

This plugin is requiring a webfont (CSS) provider and it downloads the CSS and webfont files from that provider.

I've found a post from Google Fonts that might help you with variable fonts (there's an interesting syntax: `family=Oswald:wght@200..700`):
https://fonts.google.com/knowledge/using_type/styling_type_on_the_web_with_variable_fonts

I hope it helps you.

**Comment by doutatsu** [comment]
So it has to be a webfont provider like Google? I thought it was possible to download from any URL - Inter hosts their fonts on [Cloudflare](https://cloudflare.com/), which I just download from here - https://rsms.me/inter/inter.css

Looking into the variable syntax on Google, it doesn't seem to be what I need. Comparing the two, Inter variable fonts looks like this:

```css
@font-face {
  font-family: 'Inter var';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url('font-files/Inter-roman.var.woff2?v=3.19') format('woff2');
  font-named-instance: 'Regular';
}

@font-face {
  font-family: 'Inter var';
  font-style: italic;
  font-weight: 100 900;
  font-display: swap;
  src: url('font-files/Inter-italic.var.woff2?v=3.19') format('woff2');
  font-named-instance: 'Italic';
}
```

While Google (https://fonts.googleapis.com/css2?family=Inter:wght@100..900&display=swap) doesn't have italic styles and using their "variable" format, results in this, which has a different font-family and generally doesn't seem the same to what I am looking for:

```css
/* cyrillic-ext */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2JL7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+0460-052F, U+1C80-1C88, U+20B4, U+2DE0-2DFF, U+A640-A69F, U+FE2E-FE2F;
}
/* cyrillic */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa0ZL7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+0301, U+0400-045F, U+0490-0491, U+04B0-04B1, U+2116;
}
/* greek-ext */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2ZL7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+1F00-1FFF;
}
/* greek */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1pL7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+0370-03FF;
}
/* vietnamese */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa2pL7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+0102-0103, U+0110-0111, U+0128-0129, U+0168-0169, U+01A0-01A1, U+01AF-01B0, U+0300-0301, U+0303-0304, U+0308-0309, U+0323, U+0329, U+1EA0-1EF9, U+20AB;
}
/* latin-ext */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa25L7W0Q5n-wU.woff2) format('woff2');
  unicode-range: U+0100-02AF, U+0304, U+0308, U+0329, U+1E00-1E9F, U+1EF2-1EFF, U+2020, U+20A0-20AB, U+20AD-20CF, U+2113, U+2C60-2C7F, U+A720-A7FF;
}
/* latin */
@font-face {
  font-family: 'Inter';
  font-style: normal;
  font-weight: 100 900;
  font-display: swap;
  src: url(https://fonts.gstatic.com/s/inter/v12/UcC73FwrK3iLTeHuS_fvQtMwCp50KnMa1ZL7W0Q5nw.woff2) format('woff2');
  unicode-range: U+0000-00FF, U+0131, U+0152-0153, U+02BB-02BC, U+02C6, U+02DA, U+02DC, U+0304, U+0308, U+0329, U+2000-206F, U+2074, U+20AC, U+2122, U+2191, U+2193, U+2212, U+2215, U+FEFF, U+FFFD;
}
```

**Comment by 0xb4lint** [comment]
Hi @doutatsu!

Thanks for clarifying the issue, I was able to find the problem. 😉 

The following relative URL format was not supported until now: `font-files/Inter-roman.var.woff2?v=3.19`

Could you please try it out with `v3.7.5`?

**Comment by doutatsu** [comment]
Zero config didn't work, but Simple config did!

```
<link rel="preconnect" href="https://rsms.me/">
<link rel="stylesheet" href="https://rsms.me/inter/inter.css">
...
 plugins: [
    vue(),
    UnheadVite(),
    splitVendorChunkPlugin(),
    webfontDownload(['https://rsms.me/inter/inter.css']),
    VueI18nPlugin({
      strictMessage: false,
      include: path.resolve(__dirname, './src/locales/**'),
    }),
```

<img width="404" alt="image" src="https://github.com/feat-agency/vite-plugin-webfont-dl/assets/4270980/5b6432c6-cf1e-423e-8d17-9c4de4c294e9">

And can confirm it on deployed site. Albeit, I am still getting the render blocking error on Google 🤔 
<img width="931" alt="image" src="https://github.com/feat-agency/vite-plugin-webfont-dl/assets/4270980/17415e49-dff4-4b02-a072-37cbb655838d">

**Comment by doutatsu** [comment]
Actually, is this right? Should it not point to the assets folder?
<img width="636" alt="image" src="https://github.com/feat-agency/vite-plugin-webfont-dl/assets/4270980/dfa53c8c-9c96-4c56-b025-43009cb1f596">

**Comment by doutatsu** [comment]
My bad... I actually had to write `<link rel="stylesheet" href="[CSS URL]">`, I thought CSS URL meant to be replaced with the actual URL. I am still seeing cloudfront in the HTML, but lighthouse is not complaining anymore, so I presume it is working

**Comment by 0xb4lint** [comment]
I've just released `v3.7.6` with zero-config support for rsms.me domain. 😉

**Comment by doutatsu** [comment]
haha that's amazing, I should have waited before deploying. Thanks a lot for all the help!
